src/Juvix/Compiler/Core/Translation/FromSource.hs

```diff
@@ -463,7 +463,7 @@ atoms ::
   ParsecS r Node
 atoms varsNum vars = do
   es <- P.some (atom varsNum vars)
-  return $ mkApps' (List.head es) (List.tail es)
+  return $ mkApps' (List.head es) (reverse (List.tail es))
```

Comment:
Now I'm confused. `P.some (atom varsNum vars)` parses `f t1 .. tn` into a list `[f, t1, .., tn]` (confusing expressions with corresponding nodes) so shouldn't we be calling `mkApps' f [t1, .., tn]`, without the reverse? Why did it work before?

Reply:
done